packages/web-components/src/discount-card/discount-card.ts

```diff
@@ -0,0 +1,247 @@
+import { LitElement, html } from "lit";
+import { property } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+import logger from "../utils/logger.ts";
+import { Slots } from "./constants.ts";
+import styles from "./discount-card.style.ts";
+
+/**
+ * @summary A component for displaying discount cards with various appearances
+ *
+ * @tag tapsi-discount-card
+ *
+ * @slot [action] - The slot for actions associated with the component, typically a collection of `tapsi-button` components.
+ * @slot [thumbnail] - The slot for custom image.
+ * @slot [header-icon] - The slot for header icon
+ */
+export class DiscountCard extends LitElement {
+  /** @internal */
+  public static override readonly styles = [styles];
+
+  /**
+   * The header title of the discount card.
+   *
+   * @prop {string} headerTitle
+   * @attr {string} header-title
+   * @default ""
+   */
+  @property({ attribute: "header-title" })
+  public headerTitle = "";
+
+  /**
+   * The variant of the discount card.
+   *
+   * @prop {"clay" | "whisper" | "azure" | "flame" | "grayscale" | "none"} variant
+   * @attr {"clay" | "whisper" | "azure" | "flame" | "grayscale" | "none"} variant
+   * @default "none"
+   */
+  @property({ reflect: true })
+  public variant:
+    | "clay"
+    | "whisper"
+    | "azure"
+    | "flame"
+    | "grayscale"
+    | "none" = "none";
+
+  /**
+   * The title to use for the discount-card.
+   * Represents the title of the discount-card.
```

Comment:
We can use human readable names in documentation:

```suggestion
   * The title to use for the discount card.
   * Represents the title of the discount card.
```